ql/src/java/org/apache/hadoop/hive/ql/optimizer/SharedWorkOptimizer.java

```diff
@@ -487,6 +487,15 @@ private static boolean sharedWorkOptimization(ParseContext pctx, SharedWorkOptim
             }
             LOG.debug("Input operator removed: {}", op);
           }
+
+          // A shared TSop across branches can not have probeContext that utilizes single branch info
+          // Filtered-out rows from one branch might be needed by another branch sharing a TSop
+          if (retainableTsOp.getProbeDecodeContext() != null) {
```

Comment:
In this case, should we remove the `ProbeDecodeContext` or should we skip merging these two TS operators? It may be that in some cases merging will backfire, i.e., if those two filters were very selective? Just for reference, if I remember correctly, SharedWorkOptimizer only merges TS operators targeted by SJs if at least one of the TS operators do not contain a SJ (since we would incur the full scan cost in any case at least once).

Reply:
I am not sure there is a good way to check how selective these filters are especially when using MJ HT probing (available at runtime at non-deterministic times) thus I chose to disable the optimization.

Maybe an actual decision (merging TS or not) could be made when we deal with actual static filters that can be more predictable.